### DISCUSSION TITLE
a new WebAR development course

### DISCUSSION THREAD

**Discussion by hiukim** [hiukim/mind-ar-js#58] [Ideas]
[updated on 13 Dec]

Course is now ready. Thank you for everyone support to make this happen! :)
https://www.udemy.com/course/introduction-to-web-ar-development/?referralCode=D2565F4CA6D767F30D61

----------------

Hi everyone,

I'm planning on creating a Udemy course about WebAR development to raise fund for the continuous development and support of the MindAR library. Would love to hear everyone feedback so I can better design the syllabus.

I appreciate if you could do these things:

1. comments below for your feedback. e.g. what topics are you interested/not interested in
2. share this to your network so more people can know about this
3. if you think you might be interested to enrol, please fill out the form below with your email for update: 

### Here are the tentative contents:

**Expected Learning Outcome:**
- Acquire basic understanding of the theory and algorithm behind AR
- Acquire basic knowledge on web AR related technologies, including AFRAME, three.js, webXR, model-viewer, MindAR and 8th wall.
- Gain hands-on experience on building webAR applications with MindAR opensource library and WebXR (and the knowledge should be transferable to commercial SDK like 8th wall)

**Introduction:**
- Different types of AR (image tracking, face tracking, world tracking, etc) and applications
- Different types of AR platforms (web, native apps, Facebook/instagram filters, AR glasses, etc)
- Comparison of all available web AR libraries on the market (open source and commercial)

**Theory:**
- How image tracking works (e.g. feature points detection and matching algorithms)
- How world tracking works (e.g. SLAM)
- How face tracking works (e.g. face landmark detection, machine learning)

**Practical:**
- Basic HTML, javascript, CSS and web hosting 
- AFRAME development
- THREE.js development
- WebXR
- model-viewer
- MindAR SDK
- 8th-wall SDK

**Applications:**
- Build a web AR business card for your portfolio (image tracking)
- Build a colouring card applications (turn 2D painting into 3D) (image tracking)
- Build a web virtual Try-On applications (e.g. hat, glasses, earrings, etc) (face tracking)
- AR Portal (world tracking)
- Multi-player AR games (image tracking or world tracking)

**Comment by jessicatv** [comment]
Hi I'd to talk to you about that course

**Reply by hiukim** [reply]
Sure. Any feedback is welcome! :)

**Comment by hiukim** [comment]
![Screenshot 2021-10-22 at 11 40 26 AM](https://user-images.githubusercontent.com/459126/138506615-0bcc4af1-72c4-4de1-b70c-50fbf7dcaa14.png)

Working on some interesting applications using WebAR with hand gesture interactions! 

demo video: https://www.youtube.com/watch?v=aXnqGnVRL7k

It will be a fun course! :)

**Reply by saurabhburadkar** [reply]
I am interested

**Comment by hiukim** [comment]
![Screenshot 2021-10-27 at 9 32 48 PM](https://user-images.githubusercontent.com/459126/139187425-056701b7-9667-4514-994b-ba24f0b9199c.png)

WebXR - world tracking - virtual furniture placement

demo video: https://youtu.be/P-blSkeKEl8

**Reply by HunterXuan** [reply]
> ![Screenshot 2021-10-27 at 9 32 48 PM](https://user-images.githubusercontent.com/459126/139187425-056701b7-9667-4514-994b-ba24f0b9199c.png)
> 
> WebXR - world tracking - virtual furniture placement
> 
> demo video: https://youtu.be/P-blSkeKEl8

Is this based on mind-ar-js?

**Reply by hiukim** [reply]
Nope, mind-ar-js doesn't support world tracking yet.  This one is built with WebXR

**Reply by saurabhburadkar** [reply]
I would like to learn world tracking in webAR for IOS. like XRTY, 8thwall, and zapper(instance tracking). Please consider this topic for the course.

**Comment by hiukim** [comment]
![Screenshot 2021-10-31 at 8 46 21 PM](https://user-images.githubusercontent.com/459126/139620562-af776016-ec60-431c-9c71-b4e05abf1875.png)

WebXR - AR Portals

Draw freely in the air to create an AR portal and move to another world. Support infinite number of layers.

Demo: https://www.youtube.com/watch?v=oHa3kbfSvBU

**Reply by dejaviewdujour** [reply]
I absolutely love this idea. I'm about to purchase your Udemy course. I'm wondering if this made the cut. Keep up the amazing work.

**Reply by dungnguyen234** [reply]
i have purchased your course in udemy but look like it does not have  this AR portals in the course , do you have plan for create course for this AR portal ?

**Comment by kfouani** [comment]
i would love to buy the udemy cours when its finished

**Reply by hiukim** [reply]
thx for the support!  I'm targeting December!

**Comment by hiukim** [comment]
![Screenshot 2021-11-10 at 11 04 41 PM](https://user-images.githubusercontent.com/459126/141253184-01c4358a-19c8-437b-b5bd-532dd4eee045.png)

Everything you need to know about showing video contents! Either embed a HTML5 video or stream it from Youtube/Vimeo!

I will also show you how to remove green screen to create astonishing effects!

demo: https://www.youtube.com/watch?v=rtZqtHQOScg

**Reply by littleyou-1** [reply]
You just do incredible things. I'm ready to buy your course) I also create projects based on Mind AR. Where should I send them? thank you

Ivan

**Reply by hiukim** [reply]
Thanks for the support! If you have subscribed in the above form, I'll notify you when it's ready. Otherwise, I'll also post here when it's ready! :)

What did you build? You can share a link here if you don't mind.

**Comment by raywu** [comment]
I’ve always had trouble with rendering video on AR.js via AFrame. Would
love to check this out. When wil this portion be available?

**Reply by hiukim** [reply]
Yes, there are a couple of tricky parts, especially on iOS device. 

If everything going well, it will be ready on December! :)

**Comment by hiukim** [comment]
![imgonline-com-ua-twotoone-DllC5zgqcCz](https://user-images.githubusercontent.com/459126/141421441-a636433a-c80c-48d3-8975-10b7082967c6.jpg)

Has been asked by many people. I will show you how to do a photo capture and share to social media! :)

Demo: https://www.youtube.com/watch?v=kyj8v3X7u74

**Comment by seguridadelectronica** [comment]
Waiting for the course ...  : )

**Reply by saurabhburadkar** [reply]
me too!

**Comment by EZTechnology** [comment]
Also very interested in the course.
@hiukim , could you please answer my email sent to hello@picturize.com.

**Comment by rchovatiya88** [comment]
Amazing work! super excited about the course def gone join in! Thanks!

**Comment by seguridadelectronica** [comment]
Waiting for the course ... Any advance on it?

**Reply by hiukim** [reply]
haha, yes. absolutely. 90% completed. Hopefully can finish next week. two weeks top :)

**Comment by jessicatv** [comment]
Nice!!!

On Fri, 3 Dec 2021 at 11:01, HiuKim Yuen ***@***.***> wrote:

> haha, yes. absolutely. 90% completed. Hopefully can finish next week. two
> weeks top :)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hiukim/mind-ar-js/discussions/58#discussioncomment-1743010>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAT6QTPJJFEM27DTIA42WPDUPCIQLANCNFSM5FPQH6TQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Jessica Travieso
*Chief Content Officer*


*Recognition and prestige for web designers*http://www.awwwards.com
Facebook <http://www.facebook.com/awwwards> - Twitter
<http://twitter.com/jessicaTV>
